### PR TITLE
MGMT-2227: Use running livecd when running on prem

### PR DESCRIPTION
**Pull request by djzager**
- Update ignition files to 1) copy running livecd into an accessible
  location and 2) bind mount the livecd into the assisted service as
  '/data/livecd.iso'
- Update s3wrapper to use 'livecd.iso' when looking for the base ISO
- Remove assisted-service-onprem as we should no longer need it after
  these changes
- Remove onprem references from Makefile

This allows us to unify onto a single assisted-service container image
and limit our dependence on the RHCOS livecd.

### CODE REVIEW

**Review by rwsu — APPROVED** [review]
/lgtm

A minor nit to remove unused variables.

File: pkg/s3wrapper/filesystem.go [hunk 1]
```diff
@@ -68,7 +68,7 @@ func (f *FSClient) UploadFile(ctx context.Context, filePath, objectName string)
 func (f *FSClient) UploadISO(ctx context.Context, ignitionConfig, objectPrefix string) error {
 	log := logutil.FromContext(ctx, f.log)
 	resultFile := filepath.Join(f.basedir, fmt.Sprintf("%s.iso", objectPrefix))
-	baseFile := filepath.Join(f.basedir, BaseObjectName)
+	baseFile := filepath.Join(f.basedir, "livecd.iso")
```
Comment (filanov) [inline pkg/s3wrapper/filesystem.go:71]: Maybe should be a const of configuration for FSClient?

**Comment by YuviGold** [comment]
Please remove remainders 

`skipper.yaml`
```yaml
SERVICE_ONPREM: $SERVICE_ONPREM
``` 
`docs/user-guide/assisted-service-on-local.md`
```markdown
export SERVICE_ONPREM=quay.io/<your-org>/assisted-service:latest
make build-onprem
```

**Review by filanov — APPROVED** [review]

**Review by YuviGold — APPROVED** [review]

**Pull request merged**